waveorder/stokes.py

```diff
@@ -422,3 +422,37 @@ def mmul(matrix, vector):
         raise ValueError("matrix.shape[1] is not equal to vector.shape[0]")
 
     return torch.einsum("NM...,M...->N...", matrix, vector)
+
+
+def apply_orientation_offset(orientation, rotate, flip):
+    """
+    Applies an in-place rotation and/or flip to an orientation map while
+    keeping the output range within 0 <= orientation < pi.
+
+    Parameters
+    ----------
+    orientation : array_like
+        Array of orientations measured in radians
+    rotate : bool
+        If True, rotate orientation pi/2 radians (90 degrees)
+    flip : bool
+        If True, flip the orientation
+
+    Returns
+    -------
+    array_like with same shape as input
+
+    Transformed array of orientations measured in radians
+    with range 0 <= orientation < pi
+
+    Note
+    ----
+    rotate=False and flip=False leaves the effective orientation unchanged
+    while changing the output range to 0 <= orientation < pi
+    """
+    out_orientation = torch.clone(orientation)
```

Comment:
Is this copy necessary? The docstring says that this operation is in-place.

Reply:
Thanks for the comment. I changed the docstring to
"Applies a rotation and/or flip to each voxel of an orientation map..."

My initial intent for "in-place" was to imply that the rotation is voxel-wise, but your comment made me realize that it makes it seem like I'm modifying the whole array in place. 

FWIW I found that I did need to clone if I want to use the (IMO) clean `+=` and `*=` operators. I could skip these operators, but this would end up duplicating anyway. 

I also don't want to modify the input array, because in my experience this causes tough-to-trace downstream bugs.